examples/apps/ncp/main.c

```diff
@@ -97,3 +97,24 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+
+/*
+ * Provide, if required an "otPlatLog()" function
+ */
+
+#if ((OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_NONE) ||  (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_APP_STREAM))
```

Comment:
To better understand your suggestion:

If  (OUTPUT= NONE) -- then the  openthread  core - provides a dummy otPlatLog() function that is only a stub.

If (OUTPUT = APP STREAM, better name: APPLICATION) - then  the NCP/CLI  'main.c' function provides this

If OUTPUT = PLATFORM DEFINED - then the platform must provide the 

Those three things is exactly what it does now.

Where we disagree is this:

What should be the DEFAULT for the OUTPUT - currently not all platforms provide a default implementation, those that do - I set that platform default to be "PLATFORM"   Those that don't I set the output to be NONE

To make the default PLATFORM, then we must add stubs that do nothing to various platforms because not all platforms currently provide an implementation - some provide a rich one (ie: Nordic has the RTT)  others have a stub function, and others have nothing at all.